File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.py
def solution(score):
    answer = []
    average = []
    dict_score = {}
    for i in range(len(score)) :
        average.append(score[i][0] + score[i][1])
    average.sort(reverse = True)
    for index, i in enumerate(average) :
        if i not in dict_score :
            dict_score[i] = index + 1
    
    for i in score :
        answer.append(dict_score[sum(i)])
    
    return answer

File: 프로그래머스/0/120882. 등수 매기기/등수 매기기.java
import java.util.*;

class Solution {
    public static int rank = 1;
    public static int[] solution(int[][] score) {
        List<Integer> list = new ArrayList<>();
        for (int[] s : score) {
            Integer integer = (s[0] + s[1]);
            list.add(integer);
        }

        list.sort(Collections.reverseOrder());
        System.out.println(list);

        int[] answer = new int[score.length];
        for (int i = 0; i < score.length; i++) {
            System.out.println(i);
            int standard = list.get(i);
            int count = 0;
            for (int j = 0; j < score.length; j++) {
                if (standard == score[j][0] + score[j][1]) {
                    answer[j] = rank;
                    count++;
                }
            }
            if (i != score.length - 1 && (Objects.equals(list.get(i), list.get(i + 1)))) {
                continue;
            }
            rank += count;
        }
        return answer;
    }
}